source/ios/Additions.cs

```diff
@@ -72,14 +72,17 @@ static void ConvertToNsExceptionAndAbort(object e)
 			var msg = e.ToString();
 
 			var ex = e as Exception;
-			if(ex != null) 
-				name = string.Format("{0}: {1}", ex.GetType().FullName, ex.Message);
-
+			if (ex != null) {
+				name = ex.GetType ().FullName;
+				if (ex.StackTrace != null) {
+					msg = msg.Insert (msg.IndexOf('\n'), "Xamarin Exception Stack:");
```

Comment:
I decided to get rid of the newline. But in the end it's up to you, since we strip whitespaces.